Packages/Core/Editor/Drawers/ObjectFieldSubTypeDrawer.cs

```diff
@@ -0,0 +1,41 @@
+using UnityEditor;
+using UnityEngine;
+
+namespace UnityAtoms.Editor
+{
+    /// <summary>
+    /// Property drawer To correct certain fields into the right Type. This is used in e.g. AtomList and AtomCollection.
+    /// </summary>
+    [CustomPropertyDrawer(typeof(ObjectFieldSubType))]
+    public class ObjectFieldSubTypeDrawer : PropertyDrawer
+    {
+        public override float GetPropertyHeight(SerializedProperty property,
+          GUIContent label)
+        {
+            return EditorGUI.GetPropertyHeight(property, label, true);
+        }
+
+        public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+        {
+            ObjectFieldSubType objectFieldSubType = (ObjectFieldSubType)attribute;
+
+            if (property.serializedObject.isEditingMultipleObjects)
+            {
+                EditorGUI.PropertyField(position, property, label, true);
+                return;
+            }
+
+            EditorGUI.BeginProperty(position, label, property);
+
+            position = EditorGUI.PrefixLabel(position, GUIUtility.GetControlID(FocusType.Passive), label);
+
+            var indent = EditorGUI.indentLevel;
+            EditorGUI.indentLevel = 0;
+
+            property.objectReferenceValue = EditorGUI.ObjectField(position, property.objectReferenceValue, objectFieldSubType.SubType, false);
```

Comment:
I think we might be better to use this as an internal part for the framework. For now I didn't know the best way to make this attribute internal. The AtomDrawer uses the Typecache API to determine th right Type for in the inspector. However, for AtomList and AtomCollection the defaultt type that is found for the elements in the list are AtomBaseVariableList type. This is the incorrect type because it should be AtomBaseVariable. The attribute overwrites the inspector list to show the correct type to the user.  I first thought of a general option to do it in the AtomDrawer class, but i couldn't think of a clean solution so far.